File: JavaLibrary/src/main/java/com/flat/simplex/parser/logic/block/BlockDefault.java
package com.flat.simplex.parser.logic.block;

import com.flat.simplex.lexer.Key;
import com.flat.simplex.lexer.Token;
import com.flat.simplex.parser.logic.Block;
import com.flat.simplex.parser.logic.error.Error;

public class BlockDefault extends Block {

    public BlockDefault(Block parent, Token start, Token end) {
        super(parent, start);

        Token token = start;
        Token lToken = start;
        int state = 0;
        while (token != end && token != null) {
            if (state == 0 && token.getKey() == Key.Default) {
                state = 1;
            } else if (state == 1 && token.getKey() == Key.Colon) {
                state = 2;
            } else {
                error(token, Error.unexpectedToken);
            }
            lToken = token;
            token = token.getNext();
        }
        if (state < 2) {
            error(lToken, Error.unexpectedEndOfTokens);
        }
        if (getParent() instanceof BlockSwitch) {
            ((BlockSwitch) getParent()).markDefault(this);
        } else {
            error(lToken, Error.defaultOutOfPlace);
        }
    }
}

File: SimplexApp/simplex/parser/logic/block/BlockDefault.js
var Key, BlockSwitch, Error;
const Block = require("simplex/parser/logic/Block.js");

class BlockDefault extends Block {
    constructor(parent, start, end) {
        super(parent, start);

        let token = start;
        let lToken = start;
        let state = 0;
        while (token !== end && token !== null) {
            if (state === 0 && token.getKey() === Key.Default) {
                state = 1;
            } else if (state === 1 && token.getKey() === Key.Colon) {
                state = 2;
            } else {
                this.error(token, Error.unexpectedToken);
            }
            lToken = token;
            token = token.getNext();
        }

        if (state < 2) {
            this.error(lToken, Error.unexpectedEndOfTokens);
        }
        if (this.getParent() instanceof BlockSwitch) {
            this.getParent().markDefault(this);
        } else {
            this.error(lToken, Error.defaultOutOfPlace);
        }
    }
}

module.exports = BlockDefault;
Key = require("simplex/lexer/Key.js");
BlockSwitch = require("simplex/parser/logic/block/BlockSwitch.js");
Error = require("simplex/parser/logic/error/Error.js");